src/Symfony/Component/Security/Core/User/User.php

```diff
@@ -18,7 +18,7 @@
  *
  * @author Fabien Potencier <fabien@symfony.com>
  */
-final class User implements AdvancedUserInterface
+final class User implements UserInterface, EquatableInterface
```

Comment:
Looking at `RoleInterface`; i think soft deprecating is fine. Implementing `EquatableInterface` is a new feature really. Perhaps simply revert this line?